Add unittests for invalid connection parameters

andreas-bok-sociomantic opened the pull request:
Test error string and drizzle_return_t value
for invalid host, port, user, pass, db settings

Addresses bugfix #215

andreas-bok-sociomantic committed:
Restructure connect unittest

Declare drizzle_st and drizzle_return_t
variables at global scope

Rename drizzle_return_t ret variable to driz_ret
to remove shadowing of ret defined in the
ASSERT_STREQ_ macro in yatl

andreas-bok-sociomantic committed:
Add function to test connection errors

Utilty for creation, connecting and testing

andreas-bok-sociomantic committed:
Save connection settings to local variables

Add variables host, port, user, pass, db which
hold the corresponding extracted environment values

Change drizzle_create call to use the new
variables

andreas-bok-sociomantic committed:
Add unittests for invalid connection parameters

Test error string and drizzle_return_t value
for invalid host, port, user, pass, db settings

Addresses bugfix #215

andreas-bok-sociomantic commented:
Blocked on #215

andreas-bok-sociomantic commented:
Updated and unblocked 
- Rebased on latest v5.5.x

ben-palmer-sociomantic commented on tests/unit/connect.c:
```diff
@@ -92,6 +92,35 @@ int main(int argc, char *argv[])
+    "'invalid_user'@'localhost' (using password: %s)",
+    (pass[0] == '\0') ? "NO" : "YES");
+  test_connection_error("localhost", port, "invalid_user", pass, db,
+  DRIZZLE_RETURN_HANDSHAKE_FAILED, error_expected, -1);
```
Inconsistent indentation here.

ben-palmer-sociomantic commented on tests/unit/connect.c:
```diff
@@ -92,6 +92,35 @@ int main(int argc, char *argv[])
+    -1);
+
+  // invalid user
+  sprintf(error_expected, "drizzle_check_unpack_error: Access denied for user "
```
Out of interest why are we using `sprintf` here to create the expected string rather than just hard coding it?

andreas-bok-sociomantic replied:
The pass is read from the env varr MYSQL_PASSWORD. The end of the error message is `(using password : YES )` or `(using password : NO )` depending on the value of that env var. 
However do see that using `sprintf` can be removed if we hardcode the value for `pass` in this case. 
This could be applied to the other cases as well and thus simplify the code. Seems like a better option since the connection parameters are somewhat irrelevant for these tests. 
Thus will hardcode the parameters.

ben-palmer-sociomantic commented on tests/unit/connect.c:
```diff
@@ -49,16 +49,18 @@ int main(int argc, char *argv[])
   (void)argv;
-
-  drizzle_options_st *opts = drizzle_options_create();
+  drizzle_options_st *opts;
```
Could you explain in the commit message why you are making this change as its not obvious just looking at it.

andreas-bok-sociomantic replied:
It was solely to define all variables at the top of the `main` function. But see that it is unnecessary given that the options are not used in the connection failure tests. 
Will revert this change when addressing the comments

ben-palmer-sociomantic reviewed (commented):
I had a couple of small comments.

andreas-bok-sociomantic commented:
Addressed comments by @ben-palmer-sociomantic 
- Hardcoded expected error messages
- Still copy expected error message for strings over 80 chars. But use `strcpy` instead of `sprintf`

ben-palmer-sociomantic commented:
LGTM
